agent-install/edgeNodeFiles.sh

```diff
@@ -325,8 +364,27 @@ function getHorizonPackageFiles() {
     fi
 
     if [[ $opsys == 'macos' ]]; then   #future: do this for all amd64/x86_64
-        tar --strip-components 2 -zxf $PACKAGE_NAME.tar.gz "$pkgBaseName/docker/$AGENT_IMAGE_TAR_FILE"
-        chk $? "extracting $pkgBaseName/docker/$AGENT_IMAGE_TAR_FILE from $PACKAGE_NAME.tar.gz"
+        if [[ $AGENT_IMAGES_FROM_TAR == 'true' ]]; then
+            tar --strip-components 2 -zxf $PACKAGE_NAME.tar.gz "$pkgBaseName/docker/$AGENT_IMAGE_TAR_FILE"
+            chk $? "extracting $pkgBaseName/docker/$AGENT_IMAGE_TAR_FILE from $PACKAGE_NAME.tar.gz"
+        else
+            if [[ ($PULL_REGISTRY != $DEFAULT_PULL_REGISTRY) && (-n $PULL_REGISTRY) ]]; then
```

Comment:
I'm not sure this logic works for all of the possible values of the `<edge-node-type>` arg. Instead, make the logic more straightforward:
- get PULL_REGISTRY set correct (either the default or `-r` value) near the top of script, regardless of the `<edge-node-type>` arg.
- add another variable called ALREADY_LOGGED_INTO_REGISTRY which is initially set to "false", and set to "true" in either place in the code you log in